spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/metadata/DeployerConfigurationMetadataResolver.java

```diff
@@ -58,12 +68,47 @@ public void setApplicationContext(ApplicationContext applicationContext) throws
 			throw new SkipperException("Unable to read configuration metadata", e);
 		}
 		ConfigurationMetadataRepository metadataRepository = builder.build();
-		Map<String, ConfigurationMetadataProperty> properties = metadataRepository.getAllProperties();
-		properties.entrySet().stream().forEach(e -> {
-			if (e.getKey().startsWith(KEY_PREFIX)) {
-				metadataProperties.add(e.getValue());
-			}
-		});
+		Map<String, ConfigurationMetadataGroup> groups = metadataRepository.getAllGroups();
+		// 1. go through all groups and their properties
+		// 2. filter 'spring.cloud.deployer.' properties
+		// 3. pass in only group includes, empty passes through all
+		// 4. pass in group excludes
+		// 5. same logic for properties for includes and excludes
+		// 6. what's left is white/black listed props
+		groups.values().stream()
+			.filter(g -> g.getId().startsWith(KEY_PREFIX))
+			.filter(groupEmptyOrAnyMatch(deployerProperties.getGroupIncludes()))
+			.filter(groupNoneMatch(deployerProperties.getGroupExcludes()))
+			.forEach(g -> {
+				g.getProperties().values().stream()
+					.filter(propertyEmptyOrAnyMatch(deployerProperties.getPropertyIncludes()))
```

Comment:
Does `includes` mean we should always include the properties? Not sure I understand if this means we send the property with the empty value?

Reply:
No, it just means we effectively skip match if include definitions array is empty. It's something we need to do with `anyMatch` as `noneMatch` doesn't need this trick.